kubevirt_handler/tests/test_unit.py

```diff
@@ -0,0 +1,159 @@
+
+    aggregator.assert_metric_has_tags("kubevirt_handler.vmi.vcpu_delay_seconds.count", tags=metric_tags)  # counter
+    aggregator.assert_metric_has_tags("kubevirt_handler.vmi.vcpu_seconds.count", tags=metric_tags)  # counter
+    aggregator.assert_metric_has_tags("kubevirt_handler.vmi.vcpu_wait_seconds.count", tags=metric_tags)  # counter
+
+    aggregator.assert_metric_has_tags("kubevirt_handler.workqueue.adds.count", tags=metric_tags)  # counter
+    aggregator.assert_metric_has_tags("kubevirt_handler.workqueue.depth", tags=metric_tags)  # gauge
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.longest_running_processor_seconds", tags=metric_tags
+    )  # gauge
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.queue_duration_seconds.bucket", tags=metric_tags
+    )  # histogram
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.queue_duration_seconds.sum", tags=metric_tags
+    )  # histogram
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.queue_duration_seconds.count", tags=metric_tags
+    )  # histogram
+    aggregator.assert_metric_has_tags("kubevirt_handler.workqueue.retries.count", tags=metric_tags)  # counter
+    aggregator.assert_metric_has_tags("kubevirt_handler.workqueue.unfinished_work_seconds", tags=metric_tags)  # gauge
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.work_duration_seconds.bucket", tags=metric_tags
+    )  # histogram
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.work_duration_seconds.sum", tags=metric_tags
+    )  # histogram
+    aggregator.assert_metric_has_tags(
+        "kubevirt_handler.workqueue.work_duration_seconds.count", tags=metric_tags
+    )  # histogram
+
+    aggregator.assert_all_metrics_covered()
+    aggregator.assert_metrics_using_metadata(get_metadata_metrics())
+
+
+def test_logs_warning_when_healthz_endpoint_is_missing(dd_run_check, aggregator, instance, mocker, caplog):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+    del instance["kubevirt_handler_healthz_endpoint"]
+    check = KubeVirtHandlerCheck("kubevirt_handler", {}, [instance])
+    dd_run_check(check)
+    assert (
+        "Skipping health check. Please provide a `kubevirt_handler_healthz_endpoint` to ensure the health of the KubeVirt Handler."  # noqa: E501
+        in caplog.text
+        and "WARNING" in caplog.text
+    )
+
+
+def test_emits_can_connect_one_when_service_is_up(dd_run_check, aggregator, instance, mocker):
+    mocker.patch("requests.get", wraps=mock_http_responses)
+    check = KubeVirtHandlerCheck("kubevirt_handler", {}, [instance])
+    dd_run_check(check)
+    aggregator.assert_metric(
+        "kubevirt_handler.can_connect",
+        1,
+        tags=["endpoint:https://127.0.0.1:8443/healthz", *base_tags],
+    )
+
+
+def test_emits_can_connect_zero_when_service_is_down(dd_run_check, aggregator, instance):
```

Comment:
Should we test here that no other metrics are emitted?

Reply:
We could test that in a separate test